### ISSUE TITLE
Duplicate node name "fmap_select_std" found

### ISSUE THREAD

**Issue by dmd** [nipreps/fmriprep#2687]
### What happened?

I get:

`OSError: Duplicate node name "fmap_select_std" found.`

This is single-subject. The workdir is empty, and this subject is freshly downloaded and never been run before - there is no existing derivatives directory.

### What command did you use?

```shell
/usr/bin/singularity run \
    --contain \
    --cleanenv \
    -B /tmp -B /data -B /data1 -B /data2 -B /data3 -B /n -B /cm/shared \
    /cm/shared/singularity/images/fmriprep-21.0.0.simg \
    /data/belleau_lab/E21774098 \
    /data/belleau_lab/E21774098/derivatives \
    participant \
    --fs-license-file /cm/shared/freesurfer-6.0.1/license.txt \
    --participant_label KEB08 \
    --output-spaces MNI152NLin6Asym:res-2 anat fsaverage \
    --n_cpus 4 \
    --mem-mb 16384 \
    --notrack \
    --use-aroma \
    --use-syn-sdc \
    --return-all-components \
    -vv \
    -w  /data/fmriprep-workdir/ebelleau
```

### What version of fMRIPrep are you running?

21.0.0

### How are you running fMRIPrep?

Singularity

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
last line of stdout:

220118-19:09:13,326 nipype.workflow INFO:
	 Setting-up fieldmap "auto_00001" (EstimatorType.ANAT) with <sub-KEB08_T1w.nii.gz, sub-KEB08_task-EMIST_run-01_bold.nii.gz, sub-KEB08_task-EMIST_run-02_bold.nii.gz, sub-KEB08_task-HMIST_run-01_bold.nii.gz, sub-KEB08_task-HMIST_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPOST_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPOST_run-04_bold.nii.gz, sub-KEB08_task-VOCBASELINEPRE_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPRE_run-04_bold.nii.gz, sub-KEB08_task-VOCMIXEDPOST_run-01_bold.nii.gz, sub-KEB08_task-VOCMIXEDPOST_run-03_bold.nii.gz, sub-KEB08_task-VOCMIXEDPRE_run-01_bold.nii.gz, sub-KEB08_task-VOCMIXEDPRE_run-03_bold.nii.gz>


stderr:

/opt/conda/lib/python3.8/site-packages/bids/config.py:39: FutureWarning: Setting 'extension_initial_dot' will be removed in pybids 0.16.
  warnings.warn("Setting 'extension_initial_dot' will be removed in pybids 0.16.",
Process Process-2:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/cli/workflow.py", line 118, in build_workflow
    retval["workflow"] = init_fmriprep_wf()
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/workflows/base.py", line 85, in init_fmriprep_wf
    single_subject_wf = init_single_subject_wf(subject_id)
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/workflows/base.py", line 480, in init_single_subject_wf
    workflow.connect([
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/workflows.py", line 161, in connect
    self._check_nodes(newnodes)
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/workflows.py", line 769, in _check_nodes
    raise IOError('Duplicate node name "%s" found.' % node.name)
OSError: Duplicate node name "fmap_select_std" found.
```

**Comment by dmd** [comment]
The BIDS directory consists only of an empty `derivatives` folder, `CHANGES`, `README`, `dataset_description.json`, the `sourcedata` folder, and:

```
[ebelleau@micc E21774098]$ tree sub-KEB08/
sub-KEB08/
├── anat
│   ├── sub-KEB08_FLAIR.json
│   ├── sub-KEB08_FLAIR.nii.gz
│   ├── sub-KEB08_T1w.json
│   └── sub-KEB08_T1w.nii.gz
└── func
    ├── sub-KEB08_task-EMIST_run-01_bold.json
    ├── sub-KEB08_task-EMIST_run-01_bold.nii.gz
    ├── sub-KEB08_task-EMIST_run-01_sbref.json
    ├── sub-KEB08_task-EMIST_run-01_sbref.nii.gz
    ├── sub-KEB08_task-EMIST_run-02_bold.json
    ├── sub-KEB08_task-EMIST_run-02_bold.nii.gz
    ├── sub-KEB08_task-EMIST_run-02_sbref.json
    ├── sub-KEB08_task-EMIST_run-02_sbref.nii.gz
    ├── sub-KEB08_task-HMIST_run-01_bold.json
    ├── sub-KEB08_task-HMIST_run-01_bold.nii.gz
    ├── sub-KEB08_task-HMIST_run-01_sbref.json
    ├── sub-KEB08_task-HMIST_run-01_sbref.nii.gz
    ├── sub-KEB08_task-HMIST_run-02_bold.json
    ├── sub-KEB08_task-HMIST_run-02_bold.nii.gz
    ├── sub-KEB08_task-HMIST_run-02_sbref.json
    ├── sub-KEB08_task-HMIST_run-02_sbref.nii.gz
    ├── sub-KEB08_task-rest_bold.json
    ├── sub-KEB08_task-rest_bold.nii.gz
    ├── sub-KEB08_task-rest_sbref.json
    ├── sub-KEB08_task-rest_sbref.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPOST_run-02_bold.json
    ├── sub-KEB08_task-VOCBASELINEPOST_run-02_bold.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPOST_run-02_sbref.json
    ├── sub-KEB08_task-VOCBASELINEPOST_run-02_sbref.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPOST_run-04_bold.json
    ├── sub-KEB08_task-VOCBASELINEPOST_run-04_bold.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPOST_run-04_sbref.json
    ├── sub-KEB08_task-VOCBASELINEPOST_run-04_sbref.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPRE_run-02_bold.json
    ├── sub-KEB08_task-VOCBASELINEPRE_run-02_bold.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPRE_run-02_sbref.json
    ├── sub-KEB08_task-VOCBASELINEPRE_run-02_sbref.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPRE_run-04_bold.json
    ├── sub-KEB08_task-VOCBASELINEPRE_run-04_bold.nii.gz
    ├── sub-KEB08_task-VOCBASELINEPRE_run-04_sbref.json
    ├── sub-KEB08_task-VOCBASELINEPRE_run-04_sbref.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPOST_run-01_bold.json
    ├── sub-KEB08_task-VOCMIXEDPOST_run-01_bold.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPOST_run-01_sbref.json
    ├── sub-KEB08_task-VOCMIXEDPOST_run-01_sbref.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPOST_run-03_bold.json
    ├── sub-KEB08_task-VOCMIXEDPOST_run-03_bold.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPOST_run-03_sbref.json
    ├── sub-KEB08_task-VOCMIXEDPOST_run-03_sbref.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPRE_run-01_bold.json
    ├── sub-KEB08_task-VOCMIXEDPRE_run-01_bold.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPRE_run-01_sbref.json
    ├── sub-KEB08_task-VOCMIXEDPRE_run-01_sbref.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPRE_run-03_bold.json
    ├── sub-KEB08_task-VOCMIXEDPRE_run-03_bold.nii.gz
    ├── sub-KEB08_task-VOCMIXEDPRE_run-03_sbref.json
    └── sub-KEB08_task-VOCMIXEDPRE_run-03_sbref.nii.gz
```

**Comment by effigies** [comment]
I see the issue. Looks like you have BOLD series with multiple acquisition parameters, so it's generating multiple SyN estimators. Each estimator is creating the same node. We just need to avoid duplicating that. I will push a PR momentarily.

**Comment by effigies** [comment]
Fixed in #2692.
